lib/components/checkboxrow/checkboxrow.dart

```diff
@@ -0,0 +1,201 @@
+    }
+  }
+
+  late bool enableMixedState;
+  final TUICheckBoxRowState state;
+  final Function(TUICheckBoxRowState)? onChanged;
+
+  @override
+  State<StatefulWidget> createState() => _TUICheckBoxRowState();
+}
+
+class _TUICheckBoxRowState extends State<TUICheckBoxRow> {
+  TUICheckBoxRowState state = TUICheckBoxRowState.unchecked;
+
+  @override
+  initState() {
+    super.initState();
+    state = widget.state;
+  }
+
+  _setState() {
+    setState(() {
+      if (widget.enableMixedState == true) {
+        int rawValue = state.index;
+        state = TUICheckBoxRowState.getByValue((rawValue + 1) % 3);
+      } else {
+        if (state == TUICheckBoxRowState.unchecked) {
+          state = TUICheckBoxRowState.checked;
+        } else {
+          state = TUICheckBoxRowState.unchecked;
+        }
+      }
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = TUITheme.of(context);
+    final showDescription = widget.description.isNotEmpty;
+    Widget child = Row(
+      mainAxisSize: MainAxisSize.max,
+      crossAxisAlignment: getRowCrossAxisAlignment(widget.description),
+      children: [
+        Container(
+          width: 24,
+          height: 24,
+          decoration: getBoxDecoration(context),
+          margin: getMarginIfDescription(context, widget.description),
+          child: getIcon(context),
+        ),
+        const SizedBox(
+          width: 20.0,
+        ),
+        Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            Text(
+              widget.title,
+              style: theme.typography.heading7
+                  .copyWith(fontWeight: FontWeight.w600),
+            ),
+            if (showDescription)
+              Text(
+                widget.description,
+                style: theme.typography.body7.copyWith(
+                  fontWeight: FontWeight.w400,
+                ),
+              )
+          ],
+        )
+      ],
+    );
+
+    return GestureDetector(
+      onTap: () {
+        if (widget.onChanged != null) {
+          _setState();
+          widget.onChanged?.call(state);
+        }
+      },
+      child: Container(
+        decoration: getBoxDecorationRow(context),
+        padding: const EdgeInsets.symmetric(vertical: 9.0, horizontal: 10.0),
+        child: child,
+      ),
+    );
+  }
+
+  CrossAxisAlignment getRowCrossAxisAlignment(String description) {
+    if (description.isNotEmpty) return CrossAxisAlignment.start;
+    return CrossAxisAlignment.center;
+  }
+
+  EdgeInsets getMarginIfDescription(context, String descrption) {
+    if (descrption.isNotEmpty) return const EdgeInsets.only(top: 3.0);
+    return const EdgeInsets.only(top: 0.0);
+  }
+
+  BoxDecoration getBoxDecorationRow(context) {
+    final theme = TUITheme.of(context);
+    Color backgroundColor = Colors.transparent;
+    if (widget.backgroundDark) {
+      backgroundColor = theme.colors.background;
+    }
+    return BoxDecoration(
```

Comment:
add newlines before returns and before if checks like on line 124, and other similar occurrences.